src/com/company/CircularShift.java

```diff
@@ -24,4 +24,36 @@ public String toString() {
         }
         return builder.toString();
     }
+
+    @Override
+    public int compareTo(CircularShift other) {
+        int thisCount = 0;
+        int otherCount = 0;
+        int thisIndex = this.offset;
+        int otherIndex = other.offset;
+
+        while (thisCount < this.line.getNumberOfWords() && otherCount < other.line.getNumberOfWords()) {
+
+            int compareVal = this.line.getWord(thisIndex).compareToIgnoreCase(other.line.getWord(otherIndex));
+            
+            if (compareVal > 0) {
+                return 1;
+            } else if (compareVal < 0) {
+                return -1;
+            }
+            
+            thisCount++;
+            otherCount++;
+            thisIndex = (thisIndex + 1) % this.line.getNumberOfWords();
+            otherIndex = (otherIndex + 1) % other.line.getNumberOfWords();
+        }
+
+        if (thisCount >= this.line.getNumberOfWords()) {
+            return -1;
+        } else if (otherCount >= other.line.getNumberOfWords()) {
+            return 1;
+        } else {
```

Comment:
"Both circular shifted lines are exactly the same (same words, order and number of words)"